src/managed/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.Netstandard2_0.cs

```diff
@@ -0,0 +1,841 @@
+                libraries.Add(ReadTargetLibrary(ref reader, reader.GetStringValue()));
+            }
+
+            reader.CheckEndObject();
+
+            return new Target()
+            {
+                Name = targetName,
+                Libraries = libraries
+            };
+        }
+
+        private TargetLibrary ReadTargetLibrary(ref Utf8JsonReader reader, string targetLibraryName)
+        {
+            IEnumerable<Dependency> dependencies = null;
+            List<RuntimeFile> runtimes = null;
+            List<RuntimeFile> natives = null;
+            List<string> compilations = null;
+            List<RuntimeTargetEntryStub> runtimeTargets = null;
+            List<ResourceAssembly> resources = null;
+            bool? compileOnly = null;
+
+            reader.ReadStartObject();
+
+            while (reader.Read() && reader.TokenType == JsonTokenType.PropertyName)
+            {
+                switch (reader.GetStringValue())
+                {
+                    case DependencyContextStrings.DependenciesPropertyName:
+                        dependencies = ReadTargetLibraryDependencies(ref reader);
+                        break;
+                    case DependencyContextStrings.RuntimeAssembliesKey:
+                        runtimes = ReadRuntimeFiles(ref reader);
+                        break;
+                    case DependencyContextStrings.NativeLibrariesKey:
+                        natives = ReadRuntimeFiles(ref reader);
+                        break;
+                    case DependencyContextStrings.CompileTimeAssembliesKey:
+                        compilations = ReadPropertyNames(ref reader);
+                        break;
+                    case DependencyContextStrings.RuntimeTargetsPropertyName:
+                        runtimeTargets = ReadTargetLibraryRuntimeTargets(ref reader);
+                        break;
+                    case DependencyContextStrings.ResourceAssembliesPropertyName:
+                        resources = ReadTargetLibraryResources(ref reader);
+                        break;
+                    case DependencyContextStrings.CompilationOnlyPropertyName:
+                        compileOnly = reader.ReadAsBoolean();
+                        break;
+                    default:
+                        reader.Skip();
+                        break;
+                }
+            }
+
+            reader.CheckEndObject();
+
+            return new TargetLibrary()
+            {
+                Name = targetLibraryName,
+                Dependencies = dependencies ?? Enumerable.Empty<Dependency>(),
+                Runtimes = runtimes,
+                Natives = natives,
+                Compilations = compilations,
+                RuntimeTargets = runtimeTargets,
+                Resources = resources,
+                CompileOnly = compileOnly
+            };
+        }
+
+        private IEnumerable<Dependency> ReadTargetLibraryDependencies(ref Utf8JsonReader reader)
```

Comment:
I am not sure why this API was public. I have marked it as private here.
```C#
public IEnumerable<Dependency> ReadTargetLibraryDependencies(JsonTextReader reader)
```

Reply:
Note that this is a breaking change. Previously we exposed Json.NET type via this API:
```C#
public IEnumerable<Dependency> ReadTargetLibraryDependencies(JsonTextReader reader);
```

It doesn't look like anyone is calling this API though:
- https://github.com/search?l=C%23&q=ReadTargetLibraryDependencies&type=Code
- https://apisof.net/catalog/Microsoft.Extensions.DependencyModel.DependencyContextJsonReader.ReadTargetLibraryDependencies(JsonTextReader)

Reply:
It appears this was an oversight when it was [first written](https://github.com/dotnet/cli/pull/3972/files#diff-ceddfbfdc43e20bec4e1c154cdddd637R362). Making it `private` is a good change IMO.